src/SHCore.Desktop/SHCore.cs

```diff
@@ -0,0 +1,103 @@
+    /// Exported functions from the SHCore.dll Windows library
+    /// that are available to Desktop apps only.
+    /// </content>
+    public static partial class SHCore
+    {
+        /// <summary>
+        /// Queries the dots per inch (dpi) of a display.
+        /// The values of <paramref name="dpiX"/> and <paramref name="dpiY"/> are identical.
+        /// You only need to record one of the values to determine the DPI and respond appropriately.
+        /// </summary>
+        /// <param name="hmonitor">Handle of the monitor being queried</param>
+        /// <param name="dpiType">The type of DPI being queried. Possible values are from the <see cref="MONITOR_DPI_TYPE"/> enumeration</param>
+        /// <param name="dpiX">The value of the DPI along the X axis. This value always refers to the horizontal edge, even when the screen is rotated</param>
+        /// <param name="dpiY">The value of the DPI along the Y axis. This value always refers to the vertical edge, even when the screen is rotated</param>
+        /// <returns>
+        /// This function returns one of the following values:
+        /// <list>
+        /// <item><see cref="HResult.Code.S_OK"/>: The function successfully returns the X and Y DPI values for the specified monitor</item>
+        /// <item><see cref="HResult.Code.E_INVALIDARG"/>: The handle, DPI type, or pointers passed in are not valid</item>
+        /// </list>
+        /// </returns>
+        /// <remarks>
+        /// <para>
+        /// When <see cref="MONITOR_DPI_TYPE"/> is <see cref="MONITOR_DPI_TYPE.MDT_ANGULAR_DPI"/> or <see cref="MONITOR_DPI_TYPE.MDT_RAW_DPI"/>,
+        /// the returned DPI value does not include any changes that the user made to the DPI by using the desktop scaling override slider control in Control Panel.
+        /// </para>
+        /// <para>
+        /// When you call <see cref="GetDpiForMonitor"/>, you will receive different DPI values depending on the DPI awareness of the calling application.
+        /// DPI awareness is an application-level property usually defined in the application manifest.
+        /// The following table indicates how the results will differ based on the <see cref="PROCESS_DPI_AWARENESS"/> value of your application.
+        /// </para>
+        /// <list>
+        /// <item><see cref="PROCESS_DPI_AWARENESS.PROCESS_DPI_UNAWARE"/>: 96 because the app is unaware of any other scale factors</item>
+        /// <item><see cref="PROCESS_DPI_AWARENESS.PROCESS_SYSTEM_DPI_AWARE"/>:A value set to the system DPI because the app assumes all applications use the system DPI</item>
+        /// <item><see cref="PROCESS_DPI_AWARENESS.PROCESS_PER_MONITOR_DPI_AWARE"/>: The actual DPI value set by the user for that display</item>
+        /// </list>
+        /// </remarks>
+        [DllImport(nameof(SHCore), SetLastError = true)]
+        public static extern HResult GetDpiForMonitor(IntPtr hmonitor, MONITOR_DPI_TYPE dpiType, out int dpiX, out int dpiY);
```

Comment:
>  use platform invoke to call a managed object,

I've never heard of this before. Why/how would someone use P/Invoke to call a managed object? Does that apply here?